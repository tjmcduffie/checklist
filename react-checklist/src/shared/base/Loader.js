import React, {useEffect, useState} from 'react';

const Loader = () => {
  const [periods, setPeriods] = useState('');
  let timeout;

  useEffect(() => {
    timeout = setTimeout(
      () => setPeriods(periods.length === 3 ? '' : periods + '.'),
      250
    );
    return () => clearTimeout(timeout);
  }, [periods]);

  return (
    <div>Loading{periods}</div>
  );
}

export default Loader;
