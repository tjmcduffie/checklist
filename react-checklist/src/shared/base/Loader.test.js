import Loader from './Loader';
import React from 'react';
import {create} from 'react-test-renderer';

describe('Loader', () => {
  const noop = () => {};

  it('renders without crashing', () => {
    const elem = create(<Loader />);
    elem.unmount();
  });

  it('renders predictably', () => {
    const elem = create(<Loader />);
    expect(elem.toJSON()).toMatchSnapshot();
  });
});
