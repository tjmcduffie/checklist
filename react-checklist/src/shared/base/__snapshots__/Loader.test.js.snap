// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders predictably 1`] = `
<div>
  Loading
  
</div>
`;
